File: Stack/42586.py
def solution(progresses: list, speeds: list) -> list:
    answer = []
    count = 0
    top = len(progresses) - 1
    
    # stack 처럼 사용하기 위해서.. 
    speeds.reverse()
    progresses.reverse()
    
    while progresses:
        if progresses[top] >= 100:
            count += 1
            top -= 1
            progresses.pop()
        elif count > 0:
            answer.append(count)
            count = 0
        else: 
            for i in range(top+1):
                progresses[i] += speeds[i]
                
    answer.append(count)
    return answer

solution([93, 30, 55], [1, 30, 5], [2, 1])

File: Javascript/Stack-Queue/42586.js
function solution(progresses, speeds) {
    let answer = [];
    let count = 0;
    
    while(progresses.length > 0){
        if (progresses[0] >= 100){ 
            progresses.shift();
            speeds.shift();
            count += 1; 
        } else if (count > 0) {
            answer.push(count);
            count = 0;
        } else {
            for (let i in progresses) progresses[i] += speeds[i];
        }
    }
    answer.push(count);
    return answer;
}